src/store/types.ts

```diff
@@ -1,13 +1,17 @@
 import { AppState, Action as AppAction } from './modules/app/types';
 import { NavState, Action as NavAction } from './modules/nav/types';
+import { TabState, Action as TabsAction } from './modules/tabs/types';
```

Comment:
If the actions were not all named "Action" then we could avoid aliasing them each time we import different kinds of actions

Reply:
I will address all of your concerns with the names of Props and Actions in this comment (I can't respond to your general comment for the PR).

I agree that being verbose is good, and here is why I believe that my decision to use the names I did is actually more readable than if I were to use `TabsAction` or `TabManagerOwnProps`. 

1) Typically these interfaces are being used in the context of the component or module that it belongs to. For example, in all likelihood the only time `TabManagerProps` will be referenced is inside of this component. So when you instead see `Props` you instinctively know that these are the props for the component whose file you have opened. `TabManagerProps` is, in my opinion, adding unnecessary verbosity in this situation that imo actually hinders readability, not improves it.
2)  Our linting rules specify a set length of line that is permitted before we should be splitting a single line statement across multiple lines. These long names such as `TabManagerOwnProps` make it more likely to have to split a line such as the following into multiple lines:

```
const mapStateToProps: MapStateToProps<StoreProps, OwnProps, State> = (state, ownProps) => {

// becomes

const mapStateToProps: MapStateToProps<TabManagerStoreProps, TabManagerOwnProps, State> = (state, ownProps) => {
```

And being that they are types, it's function statements like this that are more likely to suffer from having many different types and therefore being split across multiple lines. Personally I find this far more difficult to read:
```
function myFunc(
  arg1,
  arg2,
  arg3,
) {

}
```

I would far prefer that we use these shorter names and use _name aliases_ as was done in this file, and enforce the usage of these aliases when importing props, actions etc from another file outside the current module or component. I know this is largely a matter of opinion, but maybe some of these points ring true to you as well?

Reply:
PS: I can't take credit for the naming conventions `OwnProps` etc as I stole it from `react-redux` itself, but thank you for the compliment :)

If we do want to make it more verbose that they are types, I think a good solution to this would be enforcing an `I` prefix on interfaces and a `T` prefix on types. I have just been avoiding doing that because in vscode I don't typically need it as it will tell me what it is.

Reply:
It makes sense that you are mostly using it in the same context, but there are still cases where it is used elsewhere (like this file for example). I guess my concern would be that since we are aliasing it, someone might alias it another way in another file, and this could lead to some confusion because different files will call it different things. I know this is really trivial and it may be just me, but I don't usually read the imports when I'm looking through code so it would take me longer to realize that they were the same thing

Reply:
How about a solution that is the best of both worlds?

```
/** TabManager OwnProps */
interface OwnProps {}

interface StoreProps {
  activeTab: string;
}

interface DispatchProps extends DispatchProp<State> {
  setTabs(tabs: ElementTabList, activeTab: ElementTab): void;
  setActiveTab(activeTab: ElementTab): void;
  clearTabs(): void;
}

interface OwnState {}

/** TabManager props */
type Props = OwnProps & StoreProps & DispatchProps;

// Here <-----
export interface TabManagerOwnState extends OwnState {}
export type TabManagerProps = Props;
```

Reply:
sure that's fine. not sure if it's too redundant but maybe we can ask the rest of the team later!  
what's the difference between interface and type?